src/array.rs

```diff
@@ -145,6 +145,24 @@ impl<T, D> PyArray<T, D> {
         self.as_ptr() as _
     }
 
+    #[inline(always)]
+    fn check_flag(&self, flag: c_int) -> bool {
+        unsafe { (*self.as_array_ptr()).flags & flag == flag }
+    }
+
+    pub fn is_contiguous(&self) -> bool {
```

Comment:
Maybe worth adding a docstring indication that returns true if the array is either C or Fortran contiguous